packages/sdk-js/src/GrowthBook.ts

```diff
@@ -1290,18 +1290,30 @@ export class GrowthBook<
   }
 
   public fireDeferredTrackingCalls() {
+    let hasInvalidTrackingCall = false;
     this._deferredTrackingCalls.forEach((call: TrackingData) => {
       if (!call || !call.experiment || !call.result) {
-        throw new Error("Invalid tracking data");
+        this.log("Invalid deferred tracking call", { call: call });
+        hasInvalidTrackingCall = true;
+      } else {
+        this._track(call.experiment, call.result);
       }
-      this._track(call.experiment, call.result);
+      this._deferredTrackingCalls.shift();
```

Comment:
I would switch this line back to how you had it.  Just do a single `this._deferredTrackingCalls = [];` after the loop.  Now that you're not throwing inside the loop, it will always reach that line.